donkeycar/parts/dgym.py

```diff
@@ -20,28 +20,64 @@ def __init__(self, sim_path, host="127.0.0.1", port=9091, headless=0, env_name="
             if not is_exe(sim_path):
                 raise Exception("The path you provided is not an executable.")
 
-        conf["exe_path"] = sim_path
-        conf["host"] = host
-        conf["port"] = port
-        conf["guid"] = 0
-        conf["frame_skip"] = 1
-        self.env = gym.make(env_name, conf=conf)
+        gym_conf = cfg.GYM_CONF
+        gym_conf["exe_path"] = sim_path
+        gym_conf["host"] = sim_host
+        gym_conf["port"] = 9091
+        gym_conf["frame_skip"] = 1
+
+        self.env = gym.make(cfg.DONKEY_GYM_ENV_NAME, conf=gym_conf)
         self.frame = self.env.reset()
         self.action = [0.0, 0.0, 0.0]
         self.running = True
-        self.info = {'pos': (0., 0., 0.),
-                     'speed': 0,
-                     'cte': 0,
-                     'gyro': (0., 0., 0.),
-                     'accel': (0., 0., 0.),
-                     'vel': (0., 0., 0.),
-                     'lidar': []}
-        self.delay = float(delay) / 1000
-        self.record_location = record_location
-        self.record_gyroaccel = record_gyroaccel
-        self.record_velocity = record_velocity
-        self.record_lidar = record_lidar
-
+        self.info = {
+            'pos': (0., 0., 0.),
+            'cte': 0.0,
+            'speed': 0.0,
+            'forward_vel': 0.0,
+            'hit': False,
+            'gyro': (0., 0., 0.),
+            'accel': (0., 0., 0.),
+            'vel': (0., 0., 0.),
+            'odom': (0., 0., 0., 0.),
+            'lidar': [],
+            'orientation': (0., 0., 0.),
+            'last_lap_time': 0.0,
+            'lap_count': 0,
+        }
+
+        # output keys corresponding to info dict values
+        self.info_keys = {
+            'pos': ['pos/x', 'pos/y', 'pos/z'],
+            'cte': 'cte',
+            'speed': 'speed',
+            'forward_vel': 'forward_vel',
+            'hit': 'hit',
+            'gyro': ['gyro/x', 'gyro/y', 'gyro/z'],
+            'accel': ['accel/x', 'accel/y', 'accel/z'],
+            'vel': ['vel/x', 'vel/y', 'vel/z'],
+            'odom': ['odom/front_left', 'odom/front_right', 'odom/rear_left', 'odom/rear_right'],
+            'lidar': 'lidar',
+            'orientation': ['orientation/roll', 'orientation/pitch', 'orientation/yaw'],
+            'last_lap_time': 'last_lap_time',
+            'lap_count': 'lap_count',
+        }
+
+        self.output_keys = {}
+
+        try:
+            for key, val in cfg.SIM_RECORD.items():
+                if cfg.SIM_RECORD[key]:
+                    outputs_key = self.info_keys[key]
+                    if isinstance(outputs_key, list):
```

Comment:
I would prefer to not unravel these vectors/lists but to return them as such. So vel, gyro, etc will be 3-d vectors. This would keep the data better together and will also making it much cleaner in passing it through the memory and into the tub.

Reply:
Yep we could do that ! for the sake of backward compability I unpacked them, but I can easily change that

Reply:
how would you prefer the vectors ? np.arrays or lists ?

Reply:
Can you make them python lists?